nautobot/dcim/filters/__init__.py

```diff
@@ -1925,3 +1937,329 @@ def _subtree(self, queryset, name, value):
         """FilterSet method for getting Groups that are or are descended from a given ControllerManagedDeviceGroup(s)."""
         params = self.generate_query__subtree(value)
         return queryset.filter(params)
+
+
+class ModuleFilterSet(
+    LocatableModelFilterSetMixin,
+    RoleModelFilterSetMixin,
+    StatusModelFilterSetMixin,
+    TenancyModelFilterSetMixin,
+    NautobotFilterSet,
+):
+    q = SearchFilter(
+        filter_predicates={
+            "asset_tag": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "serial": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "module_type__manufacturer__name": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "module_type__model": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+        }
+    )
+    is_installed = RelatedMembershipBooleanFilter(
+        field_name="parent_module_bay",
+        label="Is installed in a module bay",
+    )
+    mac_address = MultiValueMACAddressFilter(
+        field_name="interfaces__mac_address",
+        label="MAC address",
+    )
+    manufacturer = NaturalKeyOrPKMultipleChoiceFilter(
+        field_name="module_type__manufacturer",
+        queryset=Manufacturer.objects.all(),
+        to_field_name="name",
+        label="Manufacturer (name or ID)",
+    )
+    module_type = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=ModuleType.objects.all(),
+        to_field_name="model",
+        label="Module type (model or ID)",
+    )
+    has_console_ports = RelatedMembershipBooleanFilter(
+        field_name="console_ports",
+        label="Has console ports",
+    )
+    console_ports = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=ConsolePort.objects.all(),
+        to_field_name="name",
+        label="Console Ports (name or ID)",
+    )
```

Comment:
Wonder if it'd be useful to create another shared filterset mixin for DeviceFilterSet and ModuleFilterSet to avoid the need to duplicate all of these port filters.